src/main/java/org/terasology/web/webSocket/WsHandler.java

```diff
@@ -14,37 +14,67 @@
  * limitations under the License.
  */
 
-package org.terasology.web.servlet;
+package org.terasology.web.webSocket;
 
 import java.io.IOException;
 
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonSyntaxException;
 import org.eclipse.jetty.websocket.api.Session;
 import org.eclipse.jetty.websocket.api.WebSocketAdapter;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.terasology.web.io.ActionResult;
+import org.terasology.web.io.JsonSession;
 
 /**
  * Manages one websocket session
  */
-public class EventSocket extends WebSocketAdapter {
+public class WsHandler extends WebSocketAdapter {
 
-    private static final Logger logger = LoggerFactory.getLogger(EventSocket.class);
+    private static final Logger logger = LoggerFactory.getLogger(WsHandler.class);
+    private static final Gson GSON = new GsonBuilder().disableHtmlEscaping().create();
+    private JsonSession jsonSession;
 
     @Override
     public void onWebSocketConnect(Session session) {
         super.onWebSocketConnect(session);
         logger.info("Connected: " + session.getRemoteAddress());
-        trySend("Connected!");
+        jsonSession = new JsonSession();
     }
 
     @Override
     public void onWebSocketText(String message) {
         super.onWebSocketText(message);
+        ClientMessage deserializedMessage;
+        try {
+            deserializedMessage = GSON.fromJson(message, ClientMessage.class);
```

Comment:
I think that this method would be easier to read if the variable was declared inside the try/catch block and the switch statement below would be in a separate method.